src/Nox.Generator/Application/Factories/EntityFactory.template.cs

```diff
@@ -71,7 +74,7 @@ public virtual void PartialUpdateEntity({{entity.Name}}Entity entity, Dictionary
             {{- if !IsNoxTypeCreatable key.Type || key.Type == "Guid" -}}
                 {{ continue; -}}
             {{- end }}
-        entity.{{key.Name}} = {{entity.Name}}Metadata.Create{{key.Name}}(createDto.{{key.Name}}{{if entity.IsOwnedEntity}}!{{end}});
+        entity.{{key.Name}} = {{entity.Name}}Metadata.Create{{key.Name}}(createDto.{{key.Name}}{{if entity.IsOwnedEntity}}.NonNullValue<{{keyType key}}>(){{end}});
```

Comment:
now looking to this, we can not have an owned used as a Foreign key, if this is possible in Nox.Solution that is incorrect.

@anna-naumova25  lets discuss this use case